python/src/aac/io/parser/_parse_source.py

```diff
@@ -142,39 +156,51 @@ def is_token_between_locations(token, inclusive_line_start: int, inclusive_line_
     return definitions
 
 
-def _read_file_content(arch_file: str) -> str:
+def _read_arch_file_content(arch_file: str) -> str:
     """
-    Read file content method extracts text content from the specified file.
+    Read file content method extracts text content from the specified architecture as code file.
 
     Args:
         arch_file: The file to read.
 
     Returns:
         The contents of the file as a string.
     """
-    with open(arch_file, "r") as file:
-        return file.read()
+    content = ""
+    acceptable_file_extensions = [YAML_DOCUMENT_EXTENSION, AAC_DOCUMENT_EXTENSION]
 
+    _, file_ext = path.splitext(arch_file)
+    if file_ext in acceptable_file_extensions:
+        try:
+            with open(arch_file, "r") as file:
+                content = file.read()
+        except IOError as error:
+            logging.error(f"Failed to parse {arch_file} with error {error}")
```

Comment:
This IOError is primarily for non-existent files, but given that it's the general IO exception I left the error message as a generic error message with the IOerror itself providing the error-specific information such as the file not existing.